backtester/engine/live_test.go

```diff
@@ -65,21 +64,33 @@ func TestSetupLiveDataHandler(t *testing.T) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
-	dc := &dataChecker{
-		shutdown: make(chan bool),
-	}
-	err := dc.Start()
-	assert.NoError(t, err)
+	t.Run("successful start", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			shutdown: make(chan bool),
+		}
+		err := dc.Start()
+		assert.NoError(t, err)
+
+		close(dc.shutdown)
+		dc.wg.Wait()
+	})
 
-	close(dc.shutdown)
-	dc.wg.Wait()
-	atomic.CompareAndSwapUint32(&dc.started, 0, 1)
-	err = dc.Start()
-	assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
+	t.Run("already started", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			started: 1,
+		}
+		err := dc.Start()
+		assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
+	})
 
-	var dh *dataChecker
-	err = dh.Start()
-	assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
+	t.Run("nil pointer", func(t *testing.T) {
+		t.Parallel()
+		var dc *dataChecker
+		err := dc.Start()
+		assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
```

Comment:
🐒 Optional nitpick
* Also think this can lose the subtest, which means I don't really think we need any sub-tests here at all.
* And prefer to move nil checks like this to be the first thing we do
```suggestion
		assert.ErrorIs(t, (*dataChecker)(nil).Start(), gctcommon.ErrNilPointer)
```